### PR TITLE
Fix Windows tests

### LINKED ISSUE #1758: `trackref` ordering is non-deterministic on Windows
The order of weak references to class instances in `scrapy.utils.trackref.live_refs` is not guaranteed to be the order of instantiation on Windows:

```python
# test_trackref.py

from scrapy.utils import trackref


class Foo(trackref.object_ref):
    pass

counts = {True: 0, False: 0}
for i in range(1000):
    trackref.live_refs.clear()
    o1 = Foo()
    o2 = Foo()
    counts[trackref.get_oldest("Foo") is o1] += 1

print counts
```

```powershell
PS D:\Downloads\scrapy> python .\test_trackref.py
{False: 332, True: 668}
PS D:\Downloads\scrapy> python .\test_trackref.py
{False: 331, True: 669}
PS D:\Downloads\scrapy> python .\test_trackref.py
{False: 332, True: 668}
```

If I add `time.sleep(0.001)` between the `o1=` and `o2=` lines the result is as expected (`{True: 1000}`), so I guess its due to some kind of runtime optimization by CPython?

I am not sure if this poses a problem in real world use cases. Do we rapidly instantiate tracked objects somewhere and then rely on `trackref.get_oldest`?

### PR DESCRIPTION
**Pull request by jdemaeyer**
Based on #1736.

Fixes #1758.

A couple of minor changes to the tests to make them compatible with Windows.

Also addresses two smaller issues within Scrapy:
- Small change to output foramatting of the `startproject` command. (Project name and template path were outputted as `%r` instead of `'%s'`, which screwed up paths that contain backslashes)
- In `scrapy.utils.trackref`, on Windows (only), use `time.clock` instead of `time.time` for timestamping to restore deterministic ordering of tracked references

### CODE REVIEW

**Comment by jdemaeyer** [comment]
The outcome of [this assertion in the trackref tests](https://github.com/scrapy/scrapy/blob/8a25bacdcabe54115e6c5a2151e06b62029fe188/tests/test_utils_trackref.py#L59) is not deterministic. Sometimes `trackref.get_oldest` returns `o1`, sometimes it returns `o3`.

**Comment by nyov** [comment]
Hm, I would rebase this without the appveyor patch here. If these fixes are a requirement for the appveyor patch, rebase that (#1736) on top of the fixes here, instead, once they're finalized?

File: scrapy/utils/trackref.py [hunk 1]
```diff
@@ -10,12 +10,17 @@
...
 import weakref
-from time import time
 from operator import itemgetter
 from collections import defaultdict
 import six
 
+if platform.system() == "Windows":
+    make_timestamp = time.clock
+else:
+    make_timestamp = time.time
```
Comment (nyov) [inline scrapy/utils/trackref.py]: Could shorten this thing to
```py
import platform
from time import time, clock

if platform.system() == "Windows":
	time = clock
```

**Comment by kmike** [comment]
Test failures:

* 1. I think this one should be fixed in queuelib (I'm not sure how though). It may worth it to add AppVeyor support to queuelib first.
```
c:\projects\scrapy\.tox\py33-win\lib\shutil.py:371: PermissionError 
_____________ PickleLifoDiskQueueTest.test_nonserializable_object _____________ 
c:\projects\scrapy\.tox\py33-win\lib\site-packages\queuelib\tests\__init__.py:11: in tearDown 
    shutil.rmtree(self.tmpdir) 
c:\projects\scrapy\.tox\py33-win\lib\shutil.py:478: in rmtree 
    return _rmtree_unsafe(path, onerror) 
c:\projects\scrapy\.tox\py33-win\lib\shutil.py:373: in _rmtree_unsafe 
    onerror(os.unlink, fullname, sys.exc_info()) 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
 
path = 'c:\\users\\appveyor\\appdata\\local\\temp\\1\\queuelib-tests-j2j53v' 
onerror = <function rmtree.<locals>.onerror at 0x06E104F8> 
 
    def _rmtree_unsafe(path, onerror): 
        try: 
            if os.path.islink(path): 
                # symlinks to directories are forbidden, see bug #1669 
                raise OSError("Cannot call rmtree on a symbolic link") 
        except OSError: 
            onerror(os.path.islink, path, sys.exc_info()) 
            # can't continue even if onerror hook returns 
            return 
        names = [] 
        try: 
            names = os.listdir(path) 
        except os.error: 
            onerror(os.listdir, path, sys.exc_info()) 
        for name in names: 
            fullname = os.path.join(path, name) 
            try: 
                mode = os.lstat(fullname).st_mode 
            except os.error: 
                mode = 0 
            if stat.S_ISDIR(mode): 
                _rmtree_unsafe(fullname, onerror) 
            else: 
                try: 
>                   os.unlink(fullname) 
E                   PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'c:\\users\\appveyor\\appdata\\local\\temp\\1\\queuelib-tests-j2j53v\\tmplf9bic'
```

* 2. Here we (a) should expect os-dependent line separators and (b) fix an issue with extra item on Windows. Maybe (a) can be done by opening some file in text mode; I recall Python does line ending normalisation in some cases. Or just write it explicitly (https://docs.python.org/2/library/os.html#os.linesep).

```
C:\projects\scrapy\tests\test_exporters.py:304:  
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
C:\projects\scrapy\tests\test_exporters.py:246: in assertExportResult 
    self.assertCsvEqual(fp.getvalue(), expected) 
C:\projects\scrapy\tests\test_exporters.py:235: in assertCsvEqual 
    return self.assertEqual(csvsplit(first), csvsplit(second), msg) 
E   AssertionError: Lists differ: [['', '', '\r', ' ', '"[4', ',... != [['', '', '\r\n', ' ', '"[4', ... 
E    
E   First differing element 0: 
E   ['', '', '\r', ' ', '"[4', ',', ',', '8]"', 'John'] 
E   ['', '', '\r\n', ' ', '"[4', ',', ',', '8]"', 'John'] 
E    
E   First list contains 1 additional elements. 
E   First extra element 1: 
E   ['', '', '\r\n'] 
E    
E   - [['', '', '\r', ' ', '"[4', ',', ',', '8]"', 'John'], ['', '', '\r\n']] 
E   ?                                                     ----------------- - 
E    
E   + [['', '', '\r\n', ' ', '"[4', ',', ',', '8]"', 'John']] 
E   ?              ++ 
________________ CsvItemExporterTest.test_nonstring_types_item ________________ 
 
self = <tests.test_exporters.CsvItemExporterTest testMethod=test_nonstring_types_item> 
 
    def test_nonstring_types_item(self): 
        self.assertExportResult( 
            item=self._get_nonstring_types_item(), 
            include_headers_line=False, 
>           expected='22,False,3.14,2015-01-01 01:01:01\r\n' 
        ) 
```

* 3. These tests need more investigation, maybe there is a bug in Scrapy - output is `b'John\xa3,22\r\r\n'` while it should be `b'22,John\xc2\xa3\r\n'`:

```
_______________ CsvItemExporterTest.test_header_no_header_line ________________ 
 
self = <tests.test_exporters.CsvItemExporterTest testMethod=test_header_no_header_line> 
 
    def test_header_no_header_line(self): 
        for item in [self.i, dict(self.i)]: 
            self.assertExportResult( 
                item=item, 
                include_headers_line=False, 
>               expected=b'22,John\xc2\xa3\r\n', 
            ) 
 
C:\projects\scrapy\tests\test_exporters.py:285:  
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
C:\projects\scrapy\tests\test_exporters.py:246: in assertExportResult 
    self.assertCsvEqual(fp.getvalue(), expected) 
C:\projects\scrapy\tests\test_exporters.py:231: in assertCsvEqual 
    first = to_unicode(first) 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
 
text = b'John\xa3,22\r\r\n', encoding = 'utf-8', errors = 'strict' 
 
    def to_unicode(text, encoding=None, errors='strict'): 
        """Return the unicode representation of a bytes object `text`. If `text` 
        is already an unicode object, return it as-is.""" 
        if isinstance(text, six.text_type): 
            return text 
        if not isinstance(text, (bytes, six.text_type)): 
            raise TypeError('to_unicode must receive a bytes, str or unicode ' 
                            'object, got %s' % type(text).__name__) 
        if encoding is None: 
            encoding = 'utf-8' 
>       return text.decode(encoding, errors) 
E       UnicodeDecodeError: 'utf-8' codec can't decode byte 0xa3 in position 4: invalid start byte 
```

**Comment by cathalgarvey** [comment]
Hey @jdemaeyer - Do you think this PR is still viable, and are you interested in working on it?

**Comment by kmike** [comment]
Closing it, as relevant changes were pulled in. Thanks @jdemaeyer!

**Pull request closed without merging**